components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/MeApiServiceImpl.java

```diff
@@ -9,31 +13,86 @@
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.get(userUniqueId, userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
+
     @Override
-    public Response mePost( Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
-    }
+    public Response mePost(UserDTO body, Request request) throws NotFoundException {
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.create(body.toString(), userManager, null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();    }
+
     @Override
-    public Response mePut( Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response mePut(UserDTO body, Request request) throws NotFoundException {
+        String userUniqueId = getUserId(request);
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            MeResourceManager meResourceManager = new MeResourceManager();
+            SCIMResponse scimResponse = meResourceManager.updateWithPUT(userUniqueId, body.toString(), userManager,
+                    null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();    }
+
+    private String getUserId(Request request) {
+        Object authzUser = request.getProperty("authzUser");
+        String userUniqueId = null;
+        if (authzUser instanceof String) {
+            userUniqueId = (String) authzUser;
+        } else {
+            LOG.error("User id not found in the request.");
```

Comment:
Execution of this branch of code will lead to a null being returned from the function. But it seems that this function is being used without a null check! Ideally an error should be retuned to the end user stating that a mandatory field "authzUser" has not been sent